share/plugin/ambit.cc.template

```diff
@@ -0,0 +1,179 @@
+#endif
+
+#include <ambit/tensor.h>
+#include <ambit/helpers/psi4/integrals.h>
+#include <ambit/helpers/psi4/convert.h>
+
+namespace psi{ namespace @plugin@ {
+
+extern "C"
+int read_options(std::string name, Options &options)
+{
+    if (name == "@PLUGIN@"|| options.read_globals()) {
+        /*- The amount of information printed
+            to the output file -*/
+        options.add_int("PRINT", 1);
+        /*- Whether to compute two-electron integrals -*/
+        options.add_bool("DO_TEI", true);
+    }
+
+    return true;
+}
+
+extern "C"
+SharedWavefunction @plugin@(SharedWavefunction ref_wfn, Options &options)
+{
+    using namespace ambit;
+
+    boost::shared_ptr<Molecule> molecule = ref_wfn->molecule();
+
+    // Form basis object:
+    // Create a basis set parser object.
+    boost::shared_ptr<BasisSetParser> parser(new Gaussian94BasisSetParser());
+    // Construct a new basis set.
+    boost::shared_ptr<BasisSet> aoBasis = BasisSet::pyconstruct_orbital(molecule,
+                                             "BASIS", options.get_str("BASIS"));
+    boost::shared_ptr<BasisSet> auxBasis= BasisSet::pyconstruct_auxiliary(molecule,
+                                                 "DF_BASIS_MP2", options.get_str("DF_BASIS_MP2"),
+                                                 "RIFIT", options.get_str("BASIS"));
+    boost::shared_ptr<BasisSet> zero(BasisSet::zero_ao_basis_set());
+
+    // The integral factory oversees the creation of integral objects
+    boost::shared_ptr<IntegralFactory> integral(new IntegralFactory(aoBasis, aoBasis, aoBasis, aoBasis));
+
+    // N.B. This should be called after the basis has been built, because the geometry has not been
+    // fully initialized until this time.
+    molecule->print();
+    size_t nbf = aoBasis->nbf();
+    double nucrep = molecule->nuclear_repulsion_energy();
+    psi::outfile->Printf("\n    Nuclear repulsion energy: %16.8f\n\n", nucrep);
+
+    // Form the one-electron integral objects from the integral factory
+    boost::shared_ptr<OneBodyAOInt> sOBI(integral->ao_overlap());
+    boost::shared_ptr<OneBodyAOInt> tOBI(integral->ao_kinetic());
+    boost::shared_ptr<OneBodyAOInt> vOBI(integral->ao_potential());
+
+    // Create ambit tensors to store the one-electron integrals
+    Tensor S = Tensor::build(CoreTensor, "S", { nbf, nbf });
+    Tensor T = Tensor::build(CoreTensor, "T", { nbf, nbf });
+    Tensor V = Tensor::build(CoreTensor, "V", { nbf, nbf });
+
+    // Ambit provides helpers for computing and storing integrals.
+    helpers::psi4::integrals(*sOBI.get(), &S);
+    helpers::psi4::integrals(*tOBI.get(), &T);
+    helpers::psi4::integrals(*vOBI.get(), &V);
+
+    S.print();
+    T.print();
+    V.print();
+
+    // Now, the two-electron integrals
+    Tensor g;
+    {
+        TwoBodyAOInt* eri = integral->eri();
+        g = Tensor::build(CoreTensor, "g", { nbf, nbf, nbf, nbf });
+        helpers::psi4::integrals(*eri, &g);
+        delete eri;
+    }
```

Comment:
Yeah, Ambit only uses shared_ptr in one place otherwise things passed by reference. Plus, the shared_ptr it does use is from std and not boost. The only place boost is used is in the Python wrapping. I should probably make a note someplace that the Python interface isn't tested w.r.t. Psi4.